frontend/device/sdl/device.lua

```diff
@@ -130,45 +108,43 @@ local Emulator = Device:new{
 
 local Linux = Device:new{
     model = "Linux",
-    isDesktop = yes,
 }
 
 local Mac = Device:new{
     model = "Mac",
-    isDesktop = yes,
 }
 
 local UbuntuTouch = Device:new{
     model = "UbuntuTouch",
+    isDesktop = no,
     hasFrontlight = yes,
     home_dir = nil,
 }
 
 function Device:init()
-    local emulator = self.isEmulator
     -- allows to set a viewport via environment variable
     -- syntax is Lua table syntax, e.g. EMULATE_READER_VIEWPORT="{x=10,w=550,y=5,h=790}"
     local viewport = os.getenv("EMULATE_READER_VIEWPORT")
-    if emulator and viewport then
+    if viewport then
         self.viewport = require("ui/geometry"):new(loadstring("return " .. viewport)())
     end
 
     local touchless = os.getenv("DISABLE_TOUCH") == "1"
-    if emulator and touchless then
+    if touchless then
         self.isTouchDevice = no
     end
 
     local portrait = os.getenv("EMULATE_READER_FORCE_PORTRAIT")
-    if emulator and portrait then
+    if portrait then
         self.isAlwaysPortrait = yes
     end
 
     self.hasClipboard = yes
     self.screen = require("ffi/framebuffer_SDL2_0"):new{device = self, debug = logger.dbg}
-
-    local ok, re = pcall(self.screen.setWindowIcon, self.screen, "resources/koreader.png")
-    if not ok then logger.warn(re) end
-
+    if self.isEmulator or jit.os ~= "OSX" then
```

Comment:
Um, what?

Reply:
??. There's something I don't undertand. The intent is don't set the window icon if we're running within an app bundle on osx. That doesn't happen on the emulator even when `jit.os == "OSX"`

Reply:
Since the checks go from left to right, self.isEmulator is an oddly restrictive one to start with.

> The intent is don't set the window icon if we're running within an app bundle on osx.

Right, but why not?

Reply:
err, I give up! To be honest is not that important :p. It is not need when the app runs inside an application bundle (resources/koreader.icns) will be used instead.